go.mod

```diff
@@ -23,11 +23,12 @@ require (
 	github.com/json-iterator/go v1.1.12
 	github.com/klauspost/compress v1.18.0
 	github.com/minio/simdjson-go v0.4.5
+	github.com/open-feature/go-sdk v1.16.0
```

Comment:
This dependency makes me wonder if `openfeature` package should be a nested module. `open-feature/go-sdk` introduces a handful of dependencies that could increase the default bundle size when fetching `dd-trace-go`. WDYT @eliottness?